### PR TITLE
Add HoVer Dataset

### PR DESCRIPTION
**Pull request by abhishekkrthakur**
HoVer: A Dataset for Many-Hop Fact Extraction And Claim Verification
https://arxiv.org/abs/2011.03088

### CODE REVIEW

File: datasets/hover/README.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+### Languages
+
+[More Information Needed]
+
+## Dataset Structure
+
+### Data Instances
+
+```
```
Comment (lhoestq) [inline datasets/hover/README.md:72]: We expect here to see an example of a sample from the dataset.

Can you move the sample training set you provided below in this section please ?

The other section Data Fields expects just a brief description of the different fields of the data (claim, supporting_facts, etc.)

File: datasets/hover/hover.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    "id": datasets.Value("int32"),
+                    "uid": datasets.Value("string"),
+                    "claim": datasets.Value("string"),
+                    "supporting_facts": [
+                        {
+                            "key": datasets.Value("string"),
+                            "value": datasets.Value("int32"),
+                        }
+                    ],
+                    "label": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/hover/hover.py:57]: can you use the ClassLabel feature type here ?

for example
```
"label": datasets.ClassLabel(names=["NOT_SUPPORTED", "SUPPORTED])
```
when the label like in the test set, you need to replace the value "None" with -1

you'll have to regenerate the dataset_infos.json file after this change

File: datasets/hover/README.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+### Data Fields
+
+A sample training set is provided below
+
+```
+{'id': 14856, 'uid': 'a0cf45ea-b5cd-4c4e-9ffa-73b39ebd78ce', 'claim': 'The park at which Tivolis Koncertsal is located opened on 15 August 1843.', 'supporting_facts': [{'key': 'Tivolis Koncertsal', 'value': 0}, {'key': 'Tivoli Gardens', 'value': 1}], 'label': 'SUPPORTED', 'num_hops': 2, 'hpqa_id': '5abca1a55542993a06baf937'}
+```
+
+Please note that in test set sentence only id, uid and claim are available.
```
Comment (lhoestq) [inline datasets/hover/README.md:97]: Can you mention that you replace the missing labels in the test set with a default value ? (probably -1 if you switch to ClassLabel)

**Review by lhoestq — COMMENTED** [review]
This one is really cool thanks !

I left a few comments:

**Comment by abhishekkrthakur** [comment]
@lhoestq all comments addressed :)

File: datasets/hover/hover.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    "label": d["label"],
+                    "num_hops": d["num_hops"],
+                    "hpqa_id": d["hpqa_id"],
+                }
+            else:
+                resp = {
+                    "id": sentence_counter,
+                    "uid": d["uid"],
+                    "claim": d["claim"],
+                    "supporting_facts": [{"key": "None", "value": -1}],
```
Comment (lhoestq) [inline datasets/hover/hover.py:106]:
```suggestion
                    "supporting_facts": [],
```
an empty list does the job

**Review by lhoestq — APPROVED** [review]
Thanks !

**Comment by lhoestq** [comment]
merging since the CI is fixed on master
